core/string/ustring.cpp

```diff
@@ -1303,8 +1303,8 @@ Vector<String> String::rsplit(const String &p_splitter, bool p_allow_empty, int
 		}
 
 		if (left_edge < 0) {
-			// no more splitters, we're done
-			ret.push_back(substr(0, remaining_len));
+			// No more splitters, we're done.
+			ret.push_back(left(remaining_len));
```

Comment:
`remaining_len` starts off as `length()` and after every loop iteration, after ensuring `left_edge >= 0`, it is set to that. Therefore, it is always non-negative.